src/tests/json-stringify.ts

```diff
@@ -0,0 +1,27 @@
+import { doNotExecute, Equal, Expect } from "./utils";
+
+doNotExecute(() => {
+  const result = JSON.stringify(undefined);
+  type tests = [Expect<Equal<typeof result, string | undefined>>];
+});
+
+doNotExecute(() => {
+  const result = JSON.stringify(5 as undefined | number);
+  type tests = [Expect<Equal<typeof result, string | undefined>>];
+});
+
+doNotExecute(() => {
+  const result = JSON.stringify(undefined as any);
```

Comment:
Should result be `string` here?

Reply:
No, I don't think so. If `any` is received then we can't assume anything about the param, so we should return the broadest possible type `string | undefined` but correct me if I'm wrong.

Reply:
Agree with your logic here. Could you add a descriptive comment to that effect?